rtrlib/rtr/packets.c

```diff
@@ -1067,7 +1068,7 @@ static int rtr_update_spki_table(struct rtr_socket *rtr_socket, struct spki_tabl
 static int rtr_update_aspa_table(struct rtr_socket *rtr_socket, struct aspa_table *aspa_table,
 				 struct pdu_aspa **aspa_pdus, size_t pdus_size,
 				 struct aspa_update_operation **operations,
-				 struct aspa_update_operation **failed_operation,
+				 struct aspa_update_operation **failed_operation, size_t *failed_index,
```

Comment:
That's what the mandatory index field in the operation is for. We're using to a) keep the failed op's index and b) to make qsort stable

Reply:
I figured the index in the operation to not suffice: If an undo is necessary, I need to know at update operations to undo: exactly the ones that were performed successfully before. I can use failed_index (later used as `pdus_size`) to limit the amount of updates to undo.

Crucially, failed_index refers to the `i` "kind" of indexes that most closely relate to which updates are performed. If i used operations[i]->index, I would refer to the original data structure and possibly undo the wrong operations.